src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -370,6 +377,11 @@ where
         accessed_node: UniIndex,
         visit_children: ChildrenVisitMode,
     ) -> Result<(), OutErr> {
+        // Since we go bottom-up, we push the accessed node's children first,
+        // so that they are at the bottom of the stack. But later, in `finish_foreign_accesses`
+        // we want to start visiting these children of the accessed node first.
+        // So this method finishes by reversing the stack. This only works if
+        // the stack is empty initially.
```

Comment:
Now I am even more confused... we add them first since we want to visit them first, but later we reverse so they aren't actually first then...?

Reply:
It's a stack. Usually, the things you add first are visited last. Since we visit the accessed node first, it's children would usually go last. This is why we `reverse()`.